libraries/base/Data/Monoid.hs

```diff
@@ -16,8 +16,29 @@
 -- Stability   :  experimental
 -- Portability :  portable
 --
--- A class for monoids (types with an associative binary operation that
--- has an identity) with various general-purpose instances.
+-- A type @a@ is a 'Monoid' if it provides an associative function ('<>')
+-- that lets you combine any two values of type @a@ into one, and a neutral
+-- element (`mempty`) such that 
+--
+-- > a <> mempty == mempty <> a == a
+--
+-- ==== __Examples__
+-- >>> Sum 1 <> Sum 2 <> Sum 3 <> Sum 4 :: Sum Int
+-- Sum {getSum = 10}
+-- >>> mconcat [1,2,3,4] :: Sum Int
+-- Sum {getSum = 10}
+-- >>> mconcat [] :: Sum Int
+-- Sum {getSum = 0}
+-- >>> Product 1 <> Product 2 <> Product 3 <> Product 4 :: Product Int
```

Comment:
Similarly, we might also consider a brief sentence introducing `Product`.

Reply:
done, PTAL